guide.py

```diff
@@ -0,0 +1,434 @@
+      return
+
+    feature = models.Feature(
+        category=int(self.request.get('category')),
+        name=self.request.get('name'),
+        intent_stage=models.INTENT_NONE,
+        summary=self.request.get('summary'),
+        impl_status_chrome=models.NO_ACTIVE_DEV,
+        visibility=models.WARRANTS_ARTICLE,
+        standardization=models.EDITORS_DRAFT,
+        web_dev_views=models.DEV_NO_SIGNALS)
+    key = feature.put()
+
+    # TODO(ericbidelman): enumerate and remove only the relevant keys.
+    memcache.flush_all()
+
+    redirect_url = '/guide/edit/' + str(key.id())
+    return self.redirect(redirect_url)
+
+
+class ProcessOverview(common.ContentHandler):
+
+  def get(self, path, feature_id):
+    user = users.get_current_user()
+    if user is None:
+      # Redirect to public URL for unauthenticated users.
+      return self.redirect(self.DEFAULT_URL + '/' + feature_id)
+
+    # TODO(ericbidelman): This creates a additional call to
+    # _is_user_whitelisted() (also called in common.py), resulting in another
+    # db query.
+    if not self._is_user_whitelisted(user):
+      common.handle_401(self.request, self.response, Exception)
+      return
+
+    template_data = {
+        'overview_form': guideforms.OverviewForm(),
+        'process_json': json.dumps(
+            [stage._asdict() for stage in processes.BLINK_PROCESS]),
+        'progress_so_far': [],
+        }
+
+    f = models.Feature.get_by_id(long(feature_id))
+    if f is None:
+      return self.redirect(self.ADD_NEW_URL)
+
+    progress_so_far = []  # An unordered list of progress item strings.
+    progress_so_far = ['Explainer', 'API design']
+
+    # Provide new or populated form to template.
+    template_data.update({
+        'feature': f,
+        'feature_id': f.key().id,
+        'feature_json': json.dumps(f.format_for_template()),
+        'progress_so_far': json.dumps(progress_so_far),
+    })
+
+    self._add_common_template_values(template_data)
+    self.render(data=template_data, template_path=os.path.join(path + '.html'))
+
+
+class FeatureEditStage(common.ContentHandler):
+
+  LAUNCH_URL = '/admin/features/launch'
+
+  INTENT_PARAM = 'intent'
+  LAUNCH_PARAM = 'launch'
+
+  def __FullQualifyLink(self, param_name):
+    link = self.request.get(param_name) or None
+    if link:
+      if not link.startswith('http'):
+        link = db.Link('http://' + link)
+      else:
+        link = db.Link(link)
+    return link
+
+  def __ToInt(self, param_name):
+    param = self.request.get(param_name) or None
+    if param:
+      param = int(param)
+    return param
+
+  def __get_blink_component_from_bug(self, blink_components, bug_url):
+    if blink_components[0] == models.BlinkComponent.DEFAULT_COMPONENT and bug_url:
+      result = urlfetch.fetch(bug_url)
+      if result.status_code == 200:
+        soup = BeautifulSoup(result.content, 'html.parser')
+        components = soup.find_all(string=re.compile('^Blink'))
```

Comment:
This seems a touch fragile and also likely won't work w/ Restrict issues.  Are there alternatives here?

Reply:
I added a TODO to use the Monorail API rather than scraping.